src/helpers.js

```diff
@@ -155,3 +155,99 @@ export const isRectSelected = rect => selection => {
 
   return true;
 };
+
+export const transpose = m => m[0].map((x, i) => m.map(x => x[i]));
+
+export const convertTableNodeInArrayOfRows = tableNode => {
+  const map = TableMap.get(tableNode);
+  const rows = [];
+  for (let rowIndex = 0; rowIndex < map.height; rowIndex++) {
+    const rowCells = [];
+    const seen = {};
+
+    for (let colIndex = 0; colIndex < map.width; colIndex++) {
+      const cellPos = map.map[rowIndex * map.width + colIndex];
+      const cell = tableNode.nodeAt(cellPos);
+      const rect = map.findCell(cellPos);
+      if (seen[cellPos] || rect.top !== rowIndex) {
+        rowCells.push(null);
+        continue;
+      }
+      seen[cellPos] = true;
+
+      rowCells.push(cell);
+    }
+
+    rows.push(rowCells);
+  }
+
+  return rows;
+};
+
+export const convertArrayOfRowsInTableNode = (tableNode, tableArray) => {
+  const rowsPM = [];
+  const map = TableMap.get(tableNode);
+  for (let rowIndex = 0; rowIndex < map.height; rowIndex++) {
+    const row = tableNode.child(rowIndex);
+    const rowCells = [];
+
+    for (let colIndex = 0; colIndex < map.width; colIndex++) {
+      if (!tableArray[rowIndex][colIndex]) {
+        continue;
+      }
+      const cellPos = map.map[rowIndex * map.width + colIndex];
+
+      const cell = tableArray[rowIndex][colIndex];
+      const cellData = [
+        Object.assign({}, cell.attrs),
+        cell.content,
+        cell.marks
+      ];
+
+      const oldCell = tableNode.nodeAt(cellPos);
+      const newCell = oldCell.type.createChecked.apply(oldCell.type, cellData);
+      rowCells.push(newCell);
+    }
+
+    rowsPM.push(row.type.createChecked(row.attrs, rowCells, row.marks));
+  }
+
+  const newTable = tableNode.type.createChecked(
+    tableNode.attrs,
+    rowsPM,
+    tableNode.marks
+  );
+
+  return newTable;
+};
+
+export const moveTableRowsOrColumns = (
```

Comment:
I would rather extract common internals into a helper function, split `moveTableRowsOrColumns` into `moveRows` and `moveColumns`, where I'd reuse the common helper.